Lib/multiprocessing/shared_memory.py

```diff
@@ -293,9 +293,13 @@ def __init__(self, sequence=None, *, name=None):
             ]
             self._list_len = len(_formats)
             assert sum(len(fmt) <= 8 for fmt in _formats) == self._list_len
+            sum_allocated_bytes = 0
             self._allocated_bytes = tuple(
-                    self._alignment if fmt[-1] != "s" else int(fmt[:-1])
-                    for fmt in _formats
+                (sum_allocated_bytes := sum_allocated_bytes + (
```

Comment:
Hmm, please write it in a more readable way. For example:
```python
offset = 0
self._allocated_offsets = [0]
for fmt in _formats:
    offset += self._alignment if fmt[-1] != "s" else int(fmt[:-1])
    self._allocated_offsets.append(offset)
```

Reply:
I rather not change the type of `self._allocated_bytes` to `list` when `sequence is not None`. Compare this to the call `self._allocated_bytes = struct.unpack_from(...)` when `sequence is None` in [line 352](https://github.com/tkren/cpython/blob/fb102fe7a981b3a7fd518a83714be274532d1046/Lib/multiprocessing/shared_memory.py#L352), which will create a `tuple` for `self._allocated_bytes`.

If you do not like the iterable-expression with the walrus operator as part of the tuple constructor call, we either

1. construct multiple tuples while looping over `_formats` (feels clumsy)

1. use something like 
```python
itertools.accumulate(_formats, lambda total, fmt: total + (self._alignment if fmt[-1] != "s" else int(fmt[:-1])), initial=0)` to sum over

Reply:
Or just make it a list in all cases, which is the sanest thing to do IMHO.